pkg/api/dashboard.go

```diff
@@ -295,6 +297,26 @@ func (hs *HTTPServer) PostDashboard(c *models.ReqContext) response.Response {
 	if err := web.Bind(c.Req, &cmd); err != nil {
 		return response.Error(http.StatusBadRequest, "bad request data", err)
 	}
+
+	// Ideally, coremodel validation calls would be integrated into the web
+	// framework. But this does the job for now.
+	if cm, has := hs.CoremodelRegistry.Get("dashboard"); has {
```

Comment:
I don't especially love lookup by string for code analysis purposes. And in a context like this where we're doing a one-off of validation for dashboards within a single handler, it feels particularly awkward. For this case, either directly wire injecting the dashboard coremodel seems better (but has an issue with feature flagging, see other comment), or having a "registry" that's a struct with explicit keys for each object type.

However, this approach is friendlier to abstraction where we have something that wants to interact with all coremodels in a general way, where it may need to iterate over or lookup a particular coremodel. (This is how the k8sbridge in intent-api #44840 works, for example)

In any case, string-based map-ish lookup is the lowest friction way of doing this right now.